pkg/gameservers/controller.go

```diff
@@ -168,6 +177,24 @@ func NewController(
 	return c
 }
 
+func (c *Controller) enqueueGameServerBasedOnState(item interface{}) {
+	gs := item.(*v1alpha1.GameServer)
+
+	switch gs.Status.State {
+	case v1alpha1.GameServerStatePortAllocation,
+		v1alpha1.GameServerStateCreating,
+		v1alpha1.GameServerStateStarting:
+		c.creationWorkerQueue.Enqueue(gs)
+
+	case v1alpha1.GameServerStateUnhealthy,
```

Comment:
Did you mean for this to be `Shutdown` rather than `Unhealthy`?

The gameserver controller doesn't do anything to GameServers that are `Unhealthy`

Reply:
both, actually